main.tf

```diff
@@ -60,7 +77,7 @@ resource "aws_cloudfront_distribution" "cf" {
   aliases = ["${var.api_dns}", "${var.api_alias}"]
 
   viewer_certificate {
-    acm_certificate_arn = "${aws_acm_certificate.cert.arn}"
+    acm_certificate_arn = "${var.validate_cert ? aws_acm_certificate_validation.cert.certificate_arn : aws_acm_certificate.cert.arn}"
```

Comment:
How and why `aws_acm_certificate_validation.cert.certificate_arn` is any different than `aws_acm_certificate.cert.arn` ?

Reply:
It's the same, but when `var.validate_cert` is **false** there is no `aws_acm_certificate_validation`. And this one should be used when cert verification is needed, otherwise cloudfront distribution tries to get created before the verification and fails